### DISCUSSION TITLE
Can the Back to Top Button be displayed permanently?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2513] [Q&A] [answered]
With 7.1.0 and the implementation of the "Back to Top" button comes a downside I personally dislike: It only shows when you scroll up.
This kind of defeats the purpose of the button which is to give an easy way to get back to the top when scrolling down.

My question now is, if there is a way to make it always display when scrolling either up or down?
If not would the implementation of such a feature be a nice addition.

**Comment by squidfunk** [comment, marked as answer]
You can use some extra CSS to override the hidden state. The button is hidden to minimize accidental overlay of content and only displayed when the user intends to scroll up. IMHO that’s better UX than to always show it, but of course it’s a matter of taste.

**Reply by Andre601** [reply]
Hmm okay. Guess more CSS for me to change stuff xD

**Reply by Andre601** [reply]
Hey, I'm back.

Just want to ask, if I can customize the CSS in such a way that the button appears when you're X amounts away from the top (Scroll down a bit) and make it hidden once that area is visible again.

For example, make the button visible as soon as the Header changes the displayed text from the main title to the page title (I hope you get what I mean)

**Reply by squidfunk** [reply]
I'm afraid that's not possible with plain CSS. You would need to change the logic that hides/shows the button here:

https://github.com/squidfunk/mkdocs-material/blob/285ad883883311a72b9adf8345ca7d919ba491cb/src/assets/javascripts/components/top/index.ts#L88-L117

**Reply by facelessuser** [reply]
Can't you just do this?

```css
a.md-top.md-icon {
    opacity: 1 !important;
}
```

Am I missing something?

**Reply by facelessuser** [reply]
Never mind. It shows, but doesn't trigger jumping back to top when clicked.

**Reply by Andre601** [reply]
> Never mind. It shows, but doesn't trigger jumping back to top when clicked.

That and the fact that I don't want it to be always visible but only after the user scrolled a certain distance away from the header.

**Reply by Andre601** [reply]
> I'm afraid that's not possible with plain CSS. You would need to change the logic that hides/shows the button here:
> 
> https://github.com/squidfunk/mkdocs-material/blob/285ad883883311a72b9adf8345ca7d919ba491cb/src/assets/javascripts/components/top/index.ts#L88-L117

I already guessed that this will take some Javascript (or in this case Typescript) changes to work... And I suck at this stuff...
I probs just make the hidden state grey out the button and not have the animation thing play...
